packages/stylelint-config/rules/extras.js

```diff
@@ -0,0 +1,87 @@
+module.exports = {
+  rules: {
+    // Reason: Avoid compound selectors creation using nested ampersand SASS syntax.
+    // Main reasons why use this pattern:
+    // - Unification with design system components code style
+    // - Look up easily for the selector using fulltext search in IDEs
+    // - Better orientation in longer component's code when scrolled down
+    // - Makes `no-descending-specificity` stylelint rule work correctly
+    // - IDE's support to click through from selector usage to its source
+    // Docs: https://stylelint.io/user-guide/rules/list/selector-nested-pattern/
+    'selector-nested-pattern': [
+      '(^&:)|(^&\\.)|(^&\\[)',
+      {
+        message: 'Only pseudo class selector, attribute selector and combination with state class is allowed',
```

Comment:
Also, having this in the same order as in the regular expression would be helpful 🙂.

Reply:
Thank you, order was changed.

Reply:
I’m still seeing the order is not matching. In the regexp its goes: pseudo, state class, attribute.

Reply:
I must have missed it by mistake. It is in correct order now. Thank you for re-check.